test/shape_test.cpp

```diff
@@ -737,6 +738,22 @@ TEST_CASE(tuple)
     EXPECT(test::throws([&] { s.visit_type([](auto) {}); }));
 }
 
+TEST_CASE(return_shape_tuple)
+{
+    using migraphx::shape;
+    auto op = migraphx::make_op("@return");
+    std::vector<shape> s{{shape::float_type}, {shape::float_type}};
+    EXPECT(op.compute_shape(s).type() == shape::tuple_type);
```

Comment:
OK, this doesn't work. This shouldn't work :-) 
(And I failed to verify it, it seems ..)

Reply:
why doesn't it work ? what's the error ?

Reply:
Well, the assertion fails. 
`[2023-07-27T21:13:31.115Z] test_op_shape_test: /home/jenkins/workspace/bs_AMDMIGraphX_lw_fix_half_shape/src/shape.cpp:51: migraphx::shape_impl::shape_impl(shape::type_t): Assertion "t != shape::tuple_type" failed.
[2023-07-27T21:13:31.115Z] CMake Error at gdb/test_test_op_shape_test/run.cmake:12 (message):`

Comparing a default shape of tuple_type: its default construction would give out something: But it won't be comparable to hand-crafted shape (tuple_type) vector of size 2 or 3 or..

Reply:
`EXPECT(op.compute_shape(s) == migraphx::shape{s})` This can work i think.

Reply:
Yes, but I am not sure if it gives out a clear message that it is a tuple_type. I think testing @return beyond a point is an exercise in tautology. Thanks.

Reply:
test_name has clear message that is  meant for testing tuple type. 
IMO, we should be testing for equality of shapes instead of just checking for equality of types which is just one of the member of shape class. 

also return type of "@return" operator should guarantee that it preserves the lens and strides not just type. and we should be testing for that ideally.